### ISSUE TITLE
Add rule for link list indent

### ISSUE THREAD

**Issue by koppor** [DavidAnson/markdownlint#279]
### Good

```markdown
[hyperref]: https://ctan.org/pkg/hyperref
[listings]: https://ctan.org/pkg/listings
[minted]: https://ctan.org/pkg/minted
```

### Bad

```markdown
  [hyperref]: https://ctan.org/pkg/hyperref
  [listings]: https://ctan.org/pkg/listings
  [minted]: https://ctan.org/pkg/minted
```

### Explanation

I sometimes encounter Markdown documents where the list of links at the end of the document are idented by spaces. This should not happen.

I am not talking about link checking. Meaning: Is every link listed in the block really used in the document. This would be left for real future work ^^.

This somehow refs https://github.com/DavidAnson/markdownlint/issues/40
